src/pudl/transform/eia.py

```diff
@@ -583,82 +587,59 @@ def _boiler_generator_assn(  # noqa: C901
     Args:
         eia_transformed_dfs: a dictionary of post-transform dataframes
             representing the EIA database tables.
-        eia923_years: a list of the years of EIA 923 data that
-            should be used to infer the boiler-generator associations. By
-            default it is all the working years of data.
-        eia860_years: a list of the years of EIA 860 data that
-            should be used to infer the boiler-generator associations. By
-            default it is all the working years of data.
-        debug: If True, include columns in the returned dataframe
-            indicating by what method the individual boiler generator
-            associations were inferred.
 
     Returns:
-        eia_transformed_dfs (dict): Returns the same dictionary of dataframes
-        that was passed in, and adds a new dataframe to it representing
-        the boiler-generator associations as records containing
-        plant_id_eia, generator_id, boiler_id, and unit_id_pudl
+        A dataframe containing the boiler generator associations.
 
     Raises:
-        AssertionError: If the boiler - generator association graphs are not
-            bi-partite, meaning generators only connect to boilers, and boilers
-            only connect to generators.
-        AssertionError: If all the boilers do not end up with the same unit_id
-            each year.
-        AssertionError: If all the generators do not end up with the same
-            unit_id each year.
+        AssertionError: If the boiler - generator association graphs are not bi-partite,
+            meaning generators only connect to boilers, and boilers only connect to
+            generators.
+        AssertionError: If all the boilers do not end up with the same unit_id each
+            year.
+        AssertionError: If all the generators do not end up with the same unit_id each
+            year.
     """
+    debug = context.op_config["debug"]
+    eia_settings = context.resources.dataset_settings.eia
+    eia923_years = eia_settings.eia923.years
+    eia860_years = eia_settings.eia860.years
+
     if eia923_years is None:
         eia923_years = DataSource.from_id("eia923").working_partitions["years"]
     if eia860_years is None:
         eia860_years = DataSource.from_id("eia860").working_partitions["years"]
 
-    # if eia860_years or eia923_years are still empty, we can't compile the BGA.
-    # Return the unaltered input dictionary of dataframes instead.
     if not (eia860_years and eia923_years):
-        return eia_transformed_dfs
+        logger.warning("No EIA-860 or EIA-923 years specified. Returning input BGA.")
+        return eia_transformed_dfs["clean_boiler_generator_assn_eia860"]
```

Comment:
if there are no eia years.... would there be a `clean_boiler_generator_assn_eia860` at all?

Reply:
If we got this far and there are no EIA years, maybe it should just an error (like, no that should never happen?)

Reply:
We think that this whole "if not years and years" check should be removed.